crates/re_viewer/src/web.rs

```diff
@@ -173,38 +185,22 @@ enum EndpointCategory {
     WebEventListener,
 }
 
-fn categorize_uri(mut uri: String) -> EndpointCategory {
+fn categorize_uri(uri: &str) -> EndpointCategory {
     if uri.starts_with("http") || uri.ends_with(".rrd") {
-        EndpointCategory::HttpRrd(uri)
+        EndpointCategory::HttpRrd(uri.into())
     } else if uri.starts_with("ws:") || uri.starts_with("wss:") {
-        EndpointCategory::WebSocket(uri)
+        EndpointCategory::WebSocket(uri.into())
     } else if uri.starts_with("web_event:") {
         EndpointCategory::WebEventListener
     } else {
         // If this is sometyhing like `foo.com` we can't know what it is until we connect to it.
         // We could/should connect and see what it is, but for now we just take a wild guess instead:
         re_log::info!("Assuming WebSocket endpoint");
         if !uri.contains("://") {
-            uri = format!("{}://{uri}", re_ws_comms::PROTOCOL);
+            EndpointCategory::WebSocket(format!("{}://{uri}", re_ws_comms::PROTOCOL))
+        } else {
+            EndpointCategory::WebSocket(uri.into())
         }
-        EndpointCategory::WebSocket(uri)
-    }
-}
-
-fn get_url(info: &eframe::IntegrationInfo) -> String {
-    let mut url = String::new();
-    if let Some(param) = info.web_info.location.query_map.get("url") {
```

Comment:
So we no longer check for the `?url=…` query parameter in Rust now - what are the consequences of this?

Reply:
We do, I just moved it

Reply:
oh I see!